### ISSUE TITLE
[FEATURE-REQUEST] Support for joining on multiple columns

### ISSUE THREAD

**Issue by maxharlow** [vaexio/vaex#1283]
Would be great if [`join()`](https://vaex.readthedocs.io/en/latest/api.html#vaex.dataframe.DataFrameLocal.join) parameters `on`/`left_on`/`right_on` also accepted lists of column names, [like Pandas `join()`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.join.html).

For example:

```
import vaex

df1 = vaex.from_arrays(
    forename=['john', 'jane', 'john', 'jane', 'jimmy'],
    surname=['jones', 'jones', 'smith', 'smith', 'smith'],
    age=[22, 33, 44, 55, 66]
)
df2 = vaex.from_arrays(
    first=['helen', 'john', 'jane', 'john', 'jane'],
    last=['howard', 'jones', 'jones', 'smith', 'smith'],
    city=['london', 'brighton', 'new york', 'paris', 'tokyo']
)

print('df1')
print(df1)

print('df2')
print(df2)

joined = df1.join(df2, left_on=['forename', 'surname'], right_on=['first', 'last'], how='inner')

print('joined')
print(joined)
```

**Comment by impredicative** [comment]
I would honestly consider the non-existence of this feature to be a major blocker prohibiting any use of vaex.

**Comment by maartenbreddels** [comment]
Yes, the same for groupby, I'd love to have that.
We currently don't have any client that really needs this, if you need this soon and can help secure funding for this, that would be great.

**Comment by JovanVeljanoski** [comment]
Indeed, it would be great to have this. 
PRs are also very much welcome! :)

**Comment by kmcentush** [comment]
There are some workarounds you can employ to accomplish this, but it basically requires concatenating everything that you plan on joining on into one string, then joining on that single aggregate column, then dropping it. If you're interested I can paste a snippet next time I get the chance.

**Comment by impredicative** [comment]
@kmcentush That's true, although the workaround I am more likely to employ is to just not use Vaex, snd use something like Presto or DuckDB, etc. instead.
